File: Assets/ُScripts/UPArrowkey.cs
using UnityEngine;
using System.Collections;
using UnityEngine.UI;
using UnityEngine.EventSystems;
public class UPArrowkey : MonoBehaviour {
	public GameObject firstperson;
	public float maxSpeed = 10f;
	bool buttonHeld = false;
	CharacterController chController ;


	// Use this for initialization

	public void pressed (BaseEventData eventData)
	{
		buttonHeld = true;
	}
	public void notpressed(BaseEventData eventData)
	{
		buttonHeld = false;
	}
	void Start () {
	
	}
	
	// Update is called once per frame
	void Update () {
		if (buttonHeld)
		{
			// Do stuff
			//Debug.Log ("click");
			// Move Character
			//character.transform.Translate (0, 0, 1 * 5 * Time.deltaTime);
		//	chController= firstperson.GetComponent(CharacterController);
			chController=GetComponent<CharacterController>();
			Vector3 forward = transform.TransformDirection(Vector3.forward);
			chController.SimpleMove(forward);
		} 
		//else
			//Debug.Log ("not click");
	}
	

}

File: Assets/ُScripts/UpArrowkey.js
#pragma strict
import UnityEngine.UI;
import UnityEngine.EventSystems;
private var motor : CharacterMotor;
public var rotationSpeed = 50.0f;
public var speed = 0.05f;
public var firstperson: GameObject;
public var uparrowbutton: Button;

function ONUpClick ()
{
firstperson.transform.Translate (0, 0, 1 * 5 * Time.deltaTime);
}

//function OnGUI()
//{
//var NewRect : Rect = Rect(386,-210,50,50);
//if(NewRect.Contains(Event.current.mousePosition))
//Debug.Log("I am in");
 //if(Event.current.type==EventType.MouseDown && Event.current.button ==0)
 //{
// firstperson.transform.Translate(0,0,1*5*Time.deltaTime);
 ///}
//}
